openhands/server/listen_socket.py

```diff
@@ -92,7 +91,8 @@ async def connect(connection_id: str, environ):
             session_init_args = {**settings.__dict__, **session_init_args}
 
         git_provider_tokens = create_provider_tokens_object(providers_set)
-        if server_config.app_mode != AppMode.SAAS and user_secrets:
+        # Use user_secrets provider tokens if available
+        if user_secrets:
```

Comment:
This change affects functionality, please revert to include the `server_config.app_mode != AppMode.SAAS`